server/middleware/validationMiddleware.ts

```diff
@@ -21,6 +21,30 @@ export const findError = (errors: fieldErrors, fieldName: string) => {
   }
 }
 
+export const createSchema = <T = object>(shape: T) => zodAlwaysRefine(zObjectStrict(shape))
+
+const zObjectStrict = <T = object>(shape: T) => z.object({ _csrf: z.string().optional(), ...shape }).strict()
+
+const zodAlwaysRefine = <T extends z.ZodTypeAny>(zodType: T) =>
```

Comment:
```suggestion
/*
 * Ensure that all parts of the schema get tried and can fail before exiting schema checks - this ensures we don't have to 
 * have complicated schemas if we want to both ensure the order of fields and have all the schema validation run 
*/
const zodAlwaysRefine = <T extends z.ZodTypeAny>(zodType: T) =>
```

Reply:
added together with link to zod github issue about this for background info